pkg/admission/validator.go

```diff
@@ -0,0 +1,19 @@
+package admission
+
+import (
+	sav1 "sigs.k8s.io/service-apis/apis/v1alpha1"
+)
+
+func ValidateHTTPRoute(httpRoute sav1.HTTPRoute) (bool, string, error) {
+	if len(httpRoute.Spec.Rules) == 0 {
+		return true, "", nil
+	}
+	for _, rule := range httpRoute.Spec.Rules {
+		if len(rule.Filters) != 0 {
+			if len(rule.Filters) > 1 {
+				return false, "HTTPRules cannot contain more than one HTTPFilter", nil
+			}
+		}
+	}
```

Comment:
I think the intent here was a bit different - to ensure that we couldn't have more than one instance of a type filter defined. For example, specifying 2 RequestMirrorFilters in the same filter list should probably return an error. On the other hand, specifying a RequestMirrorFilter should be entirely possible alongside the HeaderModifier one. Even with that, it's a bit more ambiguous how implementations might want to support custom filters. Maybe there is some kind of room for the same filter to exist at different points in a sequence? What do you think @hbagdi?

Reply:
Yes would be good to get some agreement from different implementors commenting on the issue like Harry has here https://github.com/kubernetes-sigs/service-apis/issues/349#issuecomment-743319384

Reply:
Just clarifying, we can have more than one filter but only 1 of each type right?

Reply:
Correct, that's the limitation we have right now.

Reply:
Ok I just pushed a commit for this, it feels a little clunky but I think it works